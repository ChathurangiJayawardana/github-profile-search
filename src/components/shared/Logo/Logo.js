import React from 'react';
import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';

import colors from '../../../setup/colors';

const Logo = props => (
  <Container {...props}>
    <span className="logo-github">Github</span>
    <span className="logo-search">Search</span>
  </Container>
);

const Container = styled.div`
  background: ${colors.white};
  width: 415px;
  height: 81px;
  font-size: 60px;

  ${props => props.headerMode && css`
    width: 280px;
    height: 50px;
    font-size: 40px;
  `}

  .logo-github {
    font-family: 'Monaco', sans-serif;
    margin-right: 5px;
  }

  .logo-search {
    font-family: 'Raleway', sans-serif;
    font-weight: 200;
    font-style: italic;
  }
`;

Logo.propTypes = {
  headerMode: PropTypes.bool,
};

Logo.defaultProps = {
  headerMode: false,
};

export default Logo;

