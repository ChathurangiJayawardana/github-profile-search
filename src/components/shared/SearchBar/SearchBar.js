import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import SearchIcon from '../../../assets/images/search-icon.svg';
import colors from '../../../setup/colors';

class SearchBar extends React.Component {
  static propTypes = {
    onSearch: PropTypes.func,
    onChange: PropTypes.func,
  }

  static defaultProps = {
    onSearch: () => null,
    onChange: () => null,
  };

  constructor(props) {
    super(props);
    this.state = { value: '' };
  }

  onChange = (e) => {
    const { value } = e.target;
    const { onChange } = this.props;

    this.setState({ value });
    onChange(value);
  }

  onSearch = () => {
    const { value } = this.state;
    const { onSearch } = this.props;

    onSearch(value);
  }

  render() {
    return (
      <Container>
        <Field
          name="search"
          type="search"
          onChange={this.onChange}
          value={this.state.value}
        />
        <Button onClick={this.onSearch}>
          <img src={SearchIcon} alt="Search Icon" />
        </Button>
      </Container>
    );
  }
}

const Container = styled.div`
  margin-top: 13px;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const Field = styled.input`
  width: 650px;
  height: 50px;
  padding: 0px 20px;
  font-family: 'Raleway';
  font-size: 20px;
  box-shadow: 0 0 4px 0 rgba(0, 0, 0, 0.4);
  font-weight: 300;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  border: none;
`;

const Button = styled.button`
  width: 100px;
  height: 50px;
  border: none;
  cursor: pointer;
  box-shadow: 0 0 4px 0 rgba(0, 0, 0, 0.4);
  border-radius: 0px 2px 2px 0px;
  background-color: ${colors.purple}
`;

export default SearchBar;

