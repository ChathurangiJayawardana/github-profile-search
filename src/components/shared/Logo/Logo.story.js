import React from 'react';
import { storiesOf } from '@storybook/react';
import Logo from '.';

const stories = storiesOf('shared/Logo', module);

stories
  .add('default', () => <Logo />)
  .add('headerMode', () => <Logo headerMode />);

