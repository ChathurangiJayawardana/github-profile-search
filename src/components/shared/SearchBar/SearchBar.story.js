import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import SearchBar from '.';

const stories = storiesOf('components/SearchBar', module);

const props = {
  style: {},
  value: '',
  onChange: action('onChange Action'),
  onSearch: action('onSearch Action'),
};

stories
  .add('default', () => <SearchBar {...props} />);

