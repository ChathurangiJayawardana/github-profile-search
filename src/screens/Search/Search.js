import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';

import Logo from '../../components/shared/Logo';
import SearchBar from '../../components/shared/SearchBar';

import colors from '../../setup/colors';

export class Search extends React.Component {
  static propTypes = {
    history: PropTypes.func.isRequired,
  }

  constructor(props) {
    super(props);
    this.state = { value: '' };
  }

  onSearch = () => {
    const { value: user } = this.state;
    const { history } = this.props;
    history.push(`/users/${user}/repos`, { user });
  }

  onChange = value => this.setState({ value });

  render() {
    return (
      <Page>
        <Logo />
        <SearchBar
          value={this.state.value}
          onSearch={this.onSearch}
          onChange={this.onChange}
        />
      </Page>
    );
  }
}

const Page = styled.main`
  width: 100vw;
  height: 100vh;
  background: ${colors.white};
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
`;

export default withRouter(Search);

