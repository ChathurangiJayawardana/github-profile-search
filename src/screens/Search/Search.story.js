import React from 'react';
import { storiesOf } from '@storybook/react';
import { MemoryRouter } from 'react-router';
import Search from '.';

const stories = storiesOf('screens/Search', module);

stories
  .addDecorator(story => (
    <MemoryRouter initialEntries={['/']}>{story()}</MemoryRouter>
  ))
  .add('default', () => <Search />);

