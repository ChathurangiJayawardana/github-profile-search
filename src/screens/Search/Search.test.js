import React from 'react';
import { mount, shallow } from 'enzyme';

import Search from '.';
import Logo from '../../components/shared/Logo';
import SearchBar from '../../components/shared/SearchBar';

const props = {
  history: {},
};

describe('Search Screen', () => {
  describe('renders', () => {
    const router = jest.fn();
    it('render with props ', () => {
      const wrapper = shallow(<Search.WrappedComponent {...props} params={{ router }} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('content', () => {
    const router = jest.fn();
    const wrapper = mount(<Search.WrappedComponent {...props} params={{ router }} />);

    it('should contain 1 Logo ', () => {
      expect(wrapper.find(Logo)).toBeTruthy();
    });

    it('should contain 1 SearchBar', () => {
      expect(wrapper.find(SearchBar)).toBeTruthy();
    });
  });
});
