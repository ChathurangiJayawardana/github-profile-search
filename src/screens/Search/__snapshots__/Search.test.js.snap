// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Screen renders render with props  1`] = `ShallowWrapper {}`;
